agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -192,6 +194,12 @@ backhaul_manager::backhaul_manager(const config_file::sConfigSlave &config,
         LOG(ERROR) << "failed to allocate Topology Task!";
     }
     m_task_pool.add_task(topology_task);
+
+    auto ap_auto_configuration_task = std::make_shared<ApAutoConfigurationTask>(*this, cmdu_tx);
+    if (!ap_auto_configuration_task) {
+        LOG(ERROR) << "failed to allocate ap_auto_configuration_task Task!";
+    }
+    m_task_pool.add_task(ap_auto_configuration_task);
```

Comment:
Probably the task should not be added into the task pool if it failed to allocate.

Reply:
OOPS!
Fixing.